todo_list/todo_list_app/models.py

```diff
@@ -0,0 +1,22 @@
+from django.db import models
+from django.contrib.auth.models import User
+# Create your models here.
+
+
+class Category(models.Model):
+    name = models.CharField(max_length=100)
+    user = models.ForeignKey(User, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return self.name
+
+
+class Task(models.Model):
+    title = models.CharField(max_length=100)
+    description = models.TextField(blank=True,null=True)
+    due_date = models.DateField()
+    completed = models.BooleanField(default=False)
+    category = models.ForeignKey(Category, on_delete=models.CASCADE)
```

Comment:
See what else we can do other than CASCADE.

Reply:
CASCADE = Delete the object related to the foreign key
PROTECT=Django will raise an protecterror if we try to delete referenced object
SET_NULL=Django will place null at the column of foreign key if try to delete referenced object
SET_DEFAULT=Django will place defaut value in foreign key column if try to delete referenced object
SET=You replace value with your argument value when delete referenced object
DO_NOTHING=IN this case developer need to handle state using django action signals on deletion of referenced object